consensus/istanbul/backend/backend.go

```diff
@@ -646,6 +647,15 @@ func (sb *Backend) AuthorForBlock(number uint64) common.Address {
 	return common.ZeroAddress
 }
 
+// HashForBlock returns the block hash from the canonical chain for the given number.
+// Implements istanbul.Backend.HashForBlock
+func (sb *Backend) HashForBlock(number uint64) common.Hash {
```

Comment:
How about returning an error here if we can't find the hash? The calling method `generateEpochValidatorSetData` can return an error.

Reply:
Because the hash can never be zero for a known block, and we do not get an error back from the blockchain query, I've changed it to use the zero value as an error indicator at the [call site](https://github.com/celo-org/celo-blockchain/blob/c4b5fcae2e427d16b3614bd4d837f4b0af141f3c/consensus/istanbul/core/commit.go#L51-L55).